presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -162,6 +163,21 @@ public PlanWithProperties visitApply(ApplyNode node, HashComputationSet context)
             return new PlanWithProperties(node, ImmutableMap.of());
         }
 
+        public PlanWithProperties visitSequence(SequenceNode node, HashComputationSet context)
+        {
+            List<PlanNode> cteProducers = node.getCteProducers().stream()
+                    .map(c ->
+                            planAndEnforce(c, new HashComputationSet(), true, new HashComputationSet()).getNode())
+                    .collect(ImmutableList.toImmutableList());
+            PlanWithProperties primarySource = planAndEnforce(node.getPrimarySource(), new HashComputationSet(), true, new HashComputationSet());
```

Comment:
Shouldn't it respect what the primary source node expect?

Reply:
It does respect the expectations set by the primary source node. Therefore, it does not provide any parentPreference when calling planAndEnforce. This approach is consistent with the logic found in visitJoin(), where it processes the left and right children without altering the inherited preferences

Reply:
I was thinking about something like:
`PlanWithProperties primarySource = plan(node.getPrimarySource(), context)`

Reply:
Yeah i think this works too and makes more sense